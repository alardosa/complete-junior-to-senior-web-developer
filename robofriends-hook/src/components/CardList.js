import React from 'react';
import Card from './Card';

const CardList = ({ robots }) => {
    return(
        <div>
        {
            robots.map((user, i) => {
                return(
                    <Card key={i} name={robots[i].name} email={robots[i].email} id={robots[i].id} />
                );
            })
        }
        </div>
    );
}


export default CardList;