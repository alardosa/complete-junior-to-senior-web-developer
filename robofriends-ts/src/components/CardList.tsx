import React from 'react';
import Card from './Card';
import { IRobot } from '../containers/App'

interface IRobotArray {
  robots: Array<IRobot>
}

const CardList = (
  { robots }: IRobotArray
) => {
  return (
    <div>
      {
        robots.map((user, i) => {
          return (
            <Card
              key={i}
              id={robots[i].id}
              name={robots[i].name}
              email={robots[i].email}
              />
          );
        })
      }
    </div>
  );
}

export default CardList;