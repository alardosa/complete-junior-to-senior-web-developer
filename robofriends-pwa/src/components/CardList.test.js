import React from 'react';
import { shallow, mount, render } from 'enzyme';
import CardList from './CardList';

describe('CardList', () => {

  it('expect to render CardList component snapshot', () => {
    const mockRobots = [
      {
        id: 1,
        name: 'John Snow',
        username: 'JohnJohn',
        email: 'john@gmail.com'
      }
    ]
    expect(shallow(<CardList robots={mockRobots}/>)).toMatchSnapshot();
  })

})