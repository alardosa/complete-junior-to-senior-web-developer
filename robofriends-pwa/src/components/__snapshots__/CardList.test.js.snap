// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList expect to render CardList component snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robots={
        Array [
            Object {
              "email": "john@gmail.com",
              "id": 1,
              "name": "John Snow",
              "username": "JohnJohn",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          email="john@gmail.com"
          id={1}
          name="John Snow"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "email": "john@gmail.com",
          "id": 1,
          "name": "John Snow",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            email="john@gmail.com"
            id={1}
            name="John Snow"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "email": "john@gmail.com",
            "id": 1,
            "name": "John Snow",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
